lib/app/AppSettings.js

```diff
@@ -39,13 +39,36 @@ class AppSettings {
     return this
   }
 
+  setAttachedExtensions (extensions) {
+    this.attachedExtensions = extensions
+    return this
+  }
+
+  /**
+   * @param {string} extension
+   * @returns {AppSettings}
+   */
+  attachExtension (extension) {
+    if (!this.attachedExtensions) {
+      this.attachedExtensions = [extension]
```

Comment:
attachedExtensions should be a map, not an array
also dont initialize this here, when this function is called the map should already be initalized because we will have other functions that will try to access attachedExtensions as well